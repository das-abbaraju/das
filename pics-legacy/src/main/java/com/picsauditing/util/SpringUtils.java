package com.picsauditing.util;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.BeansException;
import org.springframework.beans.factory.BeanFactory;
import org.springframework.beans.factory.NoSuchBeanDefinitionException;
import org.springframework.context.ApplicationContext;
import org.springframework.context.ApplicationContextAware;
import org.springframework.context.ApplicationEvent;
import org.springframework.util.Assert;

/**
 * @author Okan Kahraman
 */
public class SpringUtils implements ApplicationContextAware {

	public static final String PermissionService = "PermissionService";
	public static final String ReportService = "ReportService";
	public static final String ReportPreferencesService = "ReportPreferencesService";
	public static final String BRAIN_TREE_SERVICE_BEAN_NAME = "BrainTreeService";
	public static final String BrainTree = "brainTree";
	public static final String COUNTRY_DAO_BEAN_NAME = "CountryDAO";
	public static final String INPUT_VALIDATOR_BEAN_NAME = "InputValidator";
	public static final String INVOICE_FEE_DAO_BEAN_NAME = "InvoiceFeeDAO";
	public static final String VAT_VALIDATOR_BEAN_NAME = "VATValidator";
	public static final String ACCOUNT_DAO = "AccountDAO";
	public static final String AM_BEST_DAO = "AmBestDAO";
	public static final String APP_PROPERTY_DAO = "AppPropertyDAO";
	public static final String AUDIT_TYPE_DAO = "AuditTypeDAO";
	public static final String BASIC_DAO = "BasicDAO";
	public static final String CONTRACTOR_ACCOUNT_DAO = "ContractorAccountDAO";
	public static final String CORRUPTION_PERCEPTION_INDEX_MAP = "CorruptionPerceptionIndexMap";
	public static final String COUNTRY_DAO = "CountryDAO";
	public static final String COUNTRY_SUBDIVISION_DAO = "CountrySubdivisionDAO";
	public static final String EMAIL_SENDER = "EmailSender";
	public static final String EMAIL_SUBSCRIPTION_DAO = "EmailSubscriptionDAO";
	public static final String FEATURE_TOGGLE = "FeatureToggle";
	public static final String FLAG_CRITERIA_DAO = "FlagCriteriaDAO";
	public static final String LANGUAGE_MODEL = "LanguageModel";
	public static final String NOTE_DAO = "NoteDAO";
	public static final String OPERATOR_ACCOUNT_DAO = "OperatorAccountDAO";
	public static final String USER_DAO = "UserDAO";
	public static final String EMAIL_REPORT_RUNNER = "EmailReportRunner";
	public static final String EMAIL_REQUEST_PUBLISHER = "EmailRequestPublisher";
	public static final String CONTRACTOR_SUBMENU_BUILDER = "ContractorSubmenuBuilder";
	public static final String URL_UTILS = "StreamContentProvider";
	public static final String EMPLOYEE_DAO = "EmployeeDAO";
	public static final String ACCOUNT_SERVICE = "AccountService";
	public static final String PRODUCT_SUBSCRIPTION_SERVICE = "ProductSubscriptionService";

	private static ApplicationContext applicationContext;


	private static final Logger logger = LoggerFactory.getLogger(SpringUtils.class);

	public static void publishEvent(ApplicationEvent event) {
		Assert.notNull(applicationContext, "ApplicationContext must not be null!");
		applicationContext.publishEvent(event);
	}

	@Override
	public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
		SpringUtils.applicationContext = applicationContext;
	}

	public static <T> T getBean(String beanRefName, Class<T> beanClass) {
		T bean = null;
		Assert.notNull(applicationContext, "ApplicationContext must not be null!");
		try {
			BeanFactory factory = applicationContext;
			bean = factory.getBean(beanRefName, beanClass);
		} catch (NoSuchBeanDefinitionException ex) {
			logger.error("No bean found with name {}", beanRefName, ex);
		}

		return bean;
	}

	@SuppressWarnings("unchecked")
	public static <T> T getBean(String beanRefName) {
		T bean = null;
		Assert.notNull(applicationContext, "ApplicationContext must not be null!");
		try {
			BeanFactory factory = applicationContext;
			bean = (T) factory.getBean(beanRefName);
		} catch (NoSuchBeanDefinitionException ex) {
			logger.error("No bean found with name {}", beanRefName, ex);
		}

		return bean;
	}

}
