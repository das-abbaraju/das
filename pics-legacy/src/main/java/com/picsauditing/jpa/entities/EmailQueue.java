package com.picsauditing.jpa.entities;

import static javax.persistence.GenerationType.IDENTITY;

import java.util.Date;

import javax.mail.Address;
import javax.mail.internet.AddressException;
import javax.mail.internet.InternetAddress;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;
import javax.persistence.Transient;

import com.picsauditing.access.Permissions;
import com.picsauditing.jpa.entities.builders.EmailQueueBuilder;
import org.codehaus.jackson.annotate.JsonAutoDetect;
import org.codehaus.jackson.annotate.JsonMethod;
import org.codehaus.jackson.annotate.JsonProperty;
import org.springframework.util.StringUtils;

import com.ibm.icu.util.StringTokenizer;
import com.picsauditing.util.Strings;

@SuppressWarnings("serial")
@JsonAutoDetect(JsonMethod.FIELD)
@Entity
@Table(name = "email_queue")
public class EmailQueue implements java.io.Serializable {
	public static final int HIGH_PRIORITY = 70;
	private int id;
	private EmailStatus status = EmailStatus.Pending;
	private String fromAddress = "";
	private String fromPassword = null;
	private String toAddresses = "";
	private String ccAddresses;
	private String bccAddresses;
	private String subject = "";
	private String body = "";
	private int priority = 20;
	private Date creationDate = new Date();
	private User createdBy;
	private Date sentDate;
	private EmailTemplate emailTemplate;
	private ContractorAccount contractorAccount;
	private boolean html = false;
	private Account bodyViewableBy;
	private Account subjectViewableBy;

	@JsonProperty
	@Id
	@GeneratedValue(strategy = IDENTITY)
	@Column(name = "emailID")
	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	@Enumerated(EnumType.STRING)
	@Column(nullable = false)
	public EmailStatus getStatus() {
		return status;
	}

	public void setStatus(EmailStatus status) {
		this.status = status;
	}

	@JsonProperty
	@Column(length = 150)
	public String getFromAddress() {
		return fromAddress;
	}

	@Transient
	public InternetAddress getFromAddress2() throws AddressException {
		if (Strings.isEmpty(fromAddress))
			return null;
		return new InternetAddress(fromAddress);
	}

	public void setFromAddress(String fromAddress) {
		this.fromAddress = fromAddress;
	}

	@Column(length = 50)
	public String getFromPassword() {
		return fromPassword;
	}

	public void setFromPassword(String fromPassword) {
		this.fromPassword = fromPassword;
	}

	@JsonProperty
	@Column(nullable = false, length = 1000)
	public String getToAddresses() {
		return toAddresses;
	}

	@Transient
	public Address[] getToAddresses2() throws AddressException {
		if (Strings.isEmpty(toAddresses))
			return null;
		return InternetAddress.parse(toAddresses);
	}

	public void setToAddresses(String toAddresses) {
		this.toAddresses = toAddresses;
	}

	@JsonProperty
	@Column(length = 1000)
	public String getCcAddresses() {
		return ccAddresses;
	}

	@Transient
	public Address[] getCcAddresses2() throws AddressException {
		if (Strings.isEmpty(ccAddresses))
			return null;
		return InternetAddress.parse(ccAddresses);
	}

	public void setCcAddresses(String ccAddresses) {
		this.ccAddresses = ccAddresses;
	}

	@JsonProperty
	@Column(length = 1000)
	public String getBccAddresses() {
		return bccAddresses;
	}

	@Transient
	public Address[] getBccAddresses2() throws AddressException {
		if (Strings.isEmpty(bccAddresses))
			return null;
		return InternetAddress.parse(bccAddresses);
	}

	public void setBccAddresses(String bccAddresses) {
		this.bccAddresses = bccAddresses;
	}

	@JsonProperty
	@Column(length = 150, nullable = false)
	public String getSubject() {
		return subject;
	}

	public void setSubject(String subject) {
		this.subject = subject;
	}

	@JsonProperty
	public String getBody() {
		return body;
	}

	public void setBody(String body) {
		this.body = body;
	}

	/**
	 * Higher priority numbers are sent first by the MailCron
	 * 
	 * @return
	 */
	@Column(nullable = false, length = 4)
	public int getPriority() {
		return priority;
	}

	public void setPriority(int priority) {
		this.priority = priority;
	}

	@Temporal(TemporalType.TIMESTAMP)
	public Date getCreationDate() {
		return creationDate;
	}

	public void setCreationDate(Date creationDate) {
		this.creationDate = creationDate;
	}

	@ManyToOne
	@JoinColumn(name = "createdBy")
	public User getCreatedBy() {
		return createdBy;
	}

	public void setCreatedBy(User createdBy) {
		this.createdBy = createdBy;
	}

	@Temporal(TemporalType.TIMESTAMP)
	public Date getSentDate() {
		return sentDate;
	}

	public void setSentDate(Date sentDate) {
		this.sentDate = sentDate;
	}

	@ManyToOne
	@JoinColumn(name = "templateID", nullable = true)
	public EmailTemplate getEmailTemplate() {
		return emailTemplate;
	}

	public void setEmailTemplate(EmailTemplate emailTemplate) {
		this.emailTemplate = emailTemplate;
	}

	@ManyToOne
	@JoinColumn(name = "conID")
	public ContractorAccount getContractorAccount() {
		return contractorAccount;
	}

	public void setContractorAccount(ContractorAccount contractorAccount) {
		this.contractorAccount = contractorAccount;
	}

	@ManyToOne
	@JoinColumn(name = "bodyViewableBy")
	public Account getBodyViewableBy() {
		return bodyViewableBy;
	}

	public void setBodyViewableBy(Account bodyViewableBy) {
		this.bodyViewableBy = bodyViewableBy;
	}

	public void setBodyViewableById(int bodyViewableBy) {
		if (bodyViewableBy == 0)
			this.bodyViewableBy = null;
		this.bodyViewableBy = new Account();
		this.bodyViewableBy.setId(bodyViewableBy);
	}

	@ManyToOne
	@JoinColumn(name = "subjectViewableBy")
	public Account getSubjectViewableBy() {
		return subjectViewableBy;
	}

	public void setSubjectViewableBy(Account subjectViewableBy) {
		this.subjectViewableBy = subjectViewableBy;
	}

	public void setSubjectViewableById(int subjectViewableByID) {
		if (subjectViewableByID == 0) {
			this.subjectViewableBy = null;
		}
		this.subjectViewableBy = new Account();
		this.subjectViewableBy.setId(subjectViewableByID);
	}

	@Override
	public int hashCode() {
		final int PRIME = 31;
		int result = 1;
		result = PRIME * result + id;
		return result;
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		final EmailQueue other = (EmailQueue) obj;
		if (id != other.id)
			return false;
		return true;
	}

	public void cleanupEmailAddresses() {
		toAddresses = cleanupEmailAddress(toAddresses);
		fromAddress = cleanupEmailAddress(fromAddress);
		ccAddresses = cleanupEmailAddress(ccAddresses);
		bccAddresses = cleanupEmailAddress(bccAddresses);
	}

	public String cleanupEmailAddress(String addresses) {
		if (addresses == null)
			return null;
		
		StringTokenizer st = new StringTokenizer(addresses, ",");
		StringBuilder sb = new StringBuilder();
		while (st.hasMoreTokens()) {
			String address = st.nextToken();

			address = StringUtils.trimAllWhitespace(address);
			address = StringUtils.trimLeadingCharacter(address, '[');
			address = StringUtils.trimTrailingCharacter(address, ']');
			address = StringUtils.delete(address, "mailto:");
			sb.append(address).append(",");
		}
		String returnAddress = StringUtils.trimTrailingCharacter(sb.toString(), ',');
		return returnAddress;
	}

	@JsonProperty
	public boolean isHtml() {
		return html;
	}

	public void setHtml(boolean html) {
		this.html = html;
	}

	public void setCriticalPriority()
	{
		priority = 100;
	}
	
	public void setVeryHighPriority()
	{
		priority = 90;
	}
	
	public void setHighPriority()
	{
		priority = HIGH_PRIORITY;
	}
	
	public void setMediumPriority()
	{
		priority = 50;
	}
	
	public void setLowPriority()
	{
		priority = 30;
	}
	
	public void setVeryLowPriority()
	{
		priority = 10;
	}

    public static EmailQueueBuilder builder() {
        return new EmailQueueBuilder();
    }
}
