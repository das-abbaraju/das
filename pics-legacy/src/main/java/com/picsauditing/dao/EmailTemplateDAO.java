package com.picsauditing.dao;

import java.util.List;

import javax.persistence.Query;

import org.springframework.transaction.annotation.Propagation;
import org.springframework.transaction.annotation.Transactional;

import com.picsauditing.jpa.entities.EmailTemplate;
import com.picsauditing.jpa.entities.ListType;

@SuppressWarnings("unchecked")
public class EmailTemplateDAO extends PicsDAO {
	@Transactional(propagation = Propagation.NESTED)
	public EmailTemplate save(EmailTemplate o) {
		if (o.getId() == 0) {
			em.persist(o);
		} else {
			o = em.merge(o);
		}
		return o;
	}

	@Transactional(propagation = Propagation.NESTED)
	public void remove(int id) {
		EmailTemplate row = find(id);
		remove(row);
	}

	@Transactional(propagation = Propagation.NESTED)
	public void remove(EmailTemplate row) {
		if (row != null) {
			em.remove(row);
		}
	}

	public EmailTemplate find(int id) {
		return em.find(EmailTemplate.class, id);
	}

	
	public List<EmailTemplate> findByAccountID(int accountID) {
		Query query = em.createQuery("FROM EmailTemplate WHERE accountID=? ORDER BY templateName");
		query.setParameter(1, accountID);
		return query.getResultList();
	}

	public List<EmailTemplate> findByAccountID(int accountID, ListType listType) {
		Query query = em.createQuery("FROM EmailTemplate WHERE accountID=? AND listType=? ORDER BY templateName");
		query.setParameter(1, accountID);
		query.setParameter(2, listType);
		return query.getResultList();
	}

	public List<EmailTemplate> findAll() {
		return (List<EmailTemplate>) super.findAll(EmailTemplate.class);
	}

}
