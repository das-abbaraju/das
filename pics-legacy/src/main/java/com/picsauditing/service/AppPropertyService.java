package com.picsauditing.service;

import com.picsauditing.dao.AppPropertyDAO;
import com.picsauditing.jpa.entities.AppProperty;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;

public class AppPropertyService {

	private static final Logger logger = LoggerFactory.getLogger(AppPropertyService.class);

	@Autowired
	private AppPropertyDAO appPropertyDao;

	public boolean emailSubscriptionsAreEnabled() {
		AppProperty enableSubscriptions = appPropertyDao.find("subscription.enable");
		return Boolean.parseBoolean(enableSubscriptions.getValue());
	}
}
