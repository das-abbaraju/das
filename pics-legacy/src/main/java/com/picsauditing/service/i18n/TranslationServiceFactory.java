package com.picsauditing.service.i18n;

import java.util.*;

import com.picsauditing.PICS.I18nCache;
import com.picsauditing.dao.jdbc.JdbcFeatureToggleProvider;
import com.picsauditing.i18n.model.logging.TranslationKeyDoNothingLogger;
import com.picsauditing.i18n.service.TranslationService;
import com.picsauditing.i18n.service.TranslationServiceAdapter;
import com.picsauditing.i18n.service.TranslationServiceProperties;
import com.picsauditing.model.i18n.*;
import com.picsauditing.toggle.FeatureToggle;
import com.picsauditing.toggle.FeatureToggleCheckerGroovy;
import com.spun.util.ObjectUtils;
import com.spun.util.persistence.Loader;

public class TranslationServiceFactory {
	private static Loader<Locale> localeProvider = ThreadLocalLocale.INSTANCE;
    private static final String environment = System.getProperty("pics.env");

	// for testing
	private static TranslationService translationService;
    private static FeatureToggle featureToggleChecker;
	private static TranslationService nonLoggingTranslationService;
    // for testing ^

    public static TranslationService getNonLoggingTranslationService() {
        if (nonLoggingTranslationService != null) {
            return nonLoggingTranslationService;
        }

        if (useTranslationServiceAdapter()) {
            ActionUsageContext context = new ActionUsageContext();
            TranslationServiceProperties.Builder propertyBuilder = new TranslationServiceProperties.Builder().context(context);
            propertyBuilder.translationUsageLogger(new TranslationKeyDoNothingLogger());
            return new TranslationServiceAdapter(propertyBuilder.build());
        }

        return I18nCache.getInstance();
    }

    public static TranslationService getTranslationService() {
        if (translationService != null) {
            return translationService;
        }

        if (useTranslationServiceAdapter()) {
            TranslationServiceProperties.Builder propertyBuilder = translationServiceProperties();
            return new TranslationServiceAdapter(propertyBuilder.build());
        }

        return I18nCache.getInstance();
    }

    // if we have to parameterize the command group name for more than TranslateCommand, this will have to be a more
    // sophisticated configuration object. For now, though, let's stay simple
    public static TranslationService getTranslationService(String translateCommandKey) {
        if (translationService != null) {
            return translationService;
        }

        if (useTranslationServiceAdapter()) {
            TranslationServiceProperties.Builder propertyBuilder = translationServiceProperties();
            propertyBuilder.translationCommandKey(translateCommandKey);
            return new TranslationServiceAdapter(propertyBuilder.build());
        }

        return I18nCache.getInstance();
    }

    private static TranslationServiceProperties.Builder translationServiceProperties() {
        ActionUsageContext context = new ActionUsageContext();
        TranslationServiceProperties.Builder propertyBuilder = new TranslationServiceProperties.Builder().context(context);

        if (logTranslationUsage()) {
            propertyBuilder.translationUsageLogger(new TranslationKeyAggregateUsageLogger());
        } else {
            propertyBuilder.translationUsageLogger(new TranslationKeyDoNothingLogger());
        }
        return propertyBuilder;
    }

    private static boolean useTranslationServiceAdapter() {
	    return featureToggle().isFeatureEnabled(FeatureToggle.TOGGLE_USE_TRANSLATION_SERVICE_ADAPTER);
	}

    private static boolean logTranslationUsage() {
        // note that this is a negative toggle
        return !featureToggle().isFeatureEnabled(FeatureToggle.TOGGLE_DISABLE_LOG_TRANSLATION_USAGE);
    }

    public static void registerTranslationService(TranslationService translationService) {
		TranslationServiceFactory.translationService = translationService;
	}

    public static void registerNonLoggingTranslationService(TranslationService translationService) {
        TranslationServiceFactory.nonLoggingTranslationService = translationService;
    }

    public static Locale getLocale() {
		try {
			return localeProvider.load();
		} catch (Exception e) {
			throw ObjectUtils.throwAsError(e);
		}
	}

    private static FeatureToggle featureToggle() {
        if (featureToggleChecker != null) {
            return featureToggleChecker;
        } else {
            // we're creating a new one each time to get it to re-evaluate "permissions" since MDO changed that to hold
            // the permissions object
            FeatureToggleCheckerGroovy featureToggleChecker = new FeatureToggleCheckerGroovy(new JdbcFeatureToggleProvider(), null);
            featureToggleChecker.addToggleVariable("env", environment());
            return featureToggleChecker;
        }
    }

    private static String environment() {
        if (environment == null) {
            return "UNKNOWN";
        } else {
            return environment;
        }
    }

}
