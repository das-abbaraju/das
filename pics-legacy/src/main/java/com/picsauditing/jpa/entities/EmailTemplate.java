package com.picsauditing.jpa.entities;

import com.picsauditing.model.i18n.TranslatableString;
import com.picsauditing.report.fields.FieldType;
import com.picsauditing.report.fields.ReportField;
import com.picsauditing.report.tables.FieldImportance;

import javax.persistence.Column;
import javax.persistence.*;
import java.util.*;

@SuppressWarnings("serial")
@Entity
@Table(name = "email_template")
public class EmailTemplate extends BaseTableRequiringLanguages implements java.io.Serializable {

	public static final int PASSWORD_RESET = 85;
	public static final int USERNAME_REMINDER = 86;
	public static final int POSSIBLE_DUPLICATE_EMAIL_TEMPLATE = 234;
	public static final int FINAL_TO_OPERATORS_EMAIL_TEMPLATE = 240;
	public static final int REGISTRATION_REQUEST_FINAL_EMAIL_TEMPLATE = 241;
	public static final int regReqLastChanceEmailTemplate = 242;
	public static final int regReqReminderEmailTemplate = 243;
	public static final int pendingFinalEmailTemplate = 244;
	public static final int pendingLastChanceEmailTemplate = 245;
	public static final int pendingReminderEmailTemplate = 246;
    public static final int FLAG_CHANGES_EMAIL_TEMPLATE = 55;
    public static final int NO_ACTION_EMAIL_TEMPLATE = 70;
    public static final int REMINDER_EMAIL_TEMPLATE = 12;
    public static final int OPEN_AUDITS_EMAIL_TEMPLATE = 6;
    public static final int IMPLEMENTATION_AUDIT_CONFIRMATION_EMAIL_TEMPLATE = 15;
    public static final int AUDIT_CONFIRMATION_EMAIL_TEMPLATE = 14;
    public static final int PQF_VERIFICATION_EMAIL_TEMPLATE = 11;
    public static final int TRIAL_ACCOUNT_APPROVAL_EMAIL_TEMPLATE = 73;
    public static final int DEACTIVATION_FOR_OPERATORS_EMAIL_TEMPLATE = 51;
    public static final int BRAIN_TREE_ERROR_EMAIL_TEMPLATE = 106;
    public static final int WELCOME_EMAIL_TEMPLATE = 2;
    public static final int SAFETY_SENSITIVE_DOWNGRADED_EMAIL_TEMPLATE = 367;
    public static final int RISK_LEVEL_DOWNGRADED_EMAIL_TEMPLATE = 159;
    public static final int CREDIT_CARD_EXPIRATION_EMAIL_TEMPLATE = 59;
    public static final int REQUEST_FOR_CLIENT_SITE_ADDITION_EMAIL_TEMPLATE = 321;
    public static final int NEW_USER_EMAIL_TEMPLATE = 5;
    public static final int PASSWORD_RESET_EMAIL_TEMPLATE = 85;
    public static final int INSURANCE_CERTIFICATE_EXPIRING_EMAIL_TEMPLATE = 10;
    public static final int CONTRACTOR_ADDED_EMAIL_TEMPLATE = 9;
    public static final int NOTICE_OF_FACILITY_REMOVAL = 47;
    public static final int CONTRACTOR_APPROVE_GC_RELATIONSHIP_EMAIL_TEMPLATE = 238;
    public static final int UPCOMING_IMPLEMENTATION_AUDIT_EMAIL_TEMPLATE = 247;

    private int accountID;
	private String templateName = "";
	private String subject;
	private String body;
	private String translatedSubject;
	private String translatedBody;
	private String recipient;
	private ListType listType;
	private boolean allowsVelocity = false;
	private boolean html = false;
	private boolean translated = false;

	private static final Set<Integer> PRIVATE_VALID_DEACTIVATED_EMAILS = Collections
			.unmodifiableSet(new HashSet<>(Arrays.asList(48, 51, 71, 85, 86)));

	@Column(nullable = false)
    @ReportField(type = FieldType.AccountID, importance = FieldImportance.Low)
	public int getAccountID() {
		return accountID;
	}

	public void setAccountID(int accountID) {
		this.accountID = accountID;
	}

	@Column(length = 150)
    @ReportField(importance = FieldImportance.Low)
	public String getSubject() {
		return subject;
	}

	public void setSubject(String subject) {
		this.subject = subject;
	}

    @ReportField(importance = FieldImportance.Low)
	public String getBody() {
		return body;
	}

	public void setBody(String body) {
		this.body = body;
	}

	@Transient
	public String getTranslatedSubject() {
		if (translatedSubject != null) {
			return translatedSubject;
		}

		return new TranslatableString(getI18nKey("translatedSubject")).toTranslatedString();
	}

	public void setTranslatedSubject(String translatedSubject) {
		this.translatedSubject = translatedSubject;
	}

    @Transient
    public String getTranslatedSubject(Locale locale) {
        return new TranslatableString(getI18nKey("translatedSubject")).toTranslatedString(locale);
    }

    @Transient
	public String getTranslatedBody() {
		if (translatedBody != null) {
			return translatedBody;
		}

		return new TranslatableString(getI18nKey("translatedBody")).toTranslatedString();
	}

	public void setTranslatedBody(String translatedBody) {
		this.translatedBody = translatedBody;
	}

    @Transient
    public String getTranslatedBody(Locale locale) {
        return new TranslatableString(getI18nKey("translatedBody")).toTranslatedString(locale);
    }

    @ReportField(importance = FieldImportance.Low)
    public String getRecipient() {
		return recipient;
	}

	public void setRecipient(String recipient) {
		this.recipient = recipient;
	}

    @ReportField()
	public String getTemplateName() {
		return templateName;
	}

	@Column(length = 50, nullable = false)
	public void setTemplateName(String templateName) {
		this.templateName = templateName;
	}

	@Enumerated(EnumType.STRING)
    @ReportField(type = FieldType.ListType)
	public ListType getListType() {
		return listType;
	}

	public void setListType(ListType listType) {
		this.listType = listType;
	}

    @ReportField(type = FieldType.Boolean)
	public boolean isAllowsVelocity() {
		return allowsVelocity;
	}

	public void setAllowsVelocity(boolean allowsVelocity) {
		this.allowsVelocity = allowsVelocity;
	}

    @ReportField(type = FieldType.Boolean)
	public boolean isHtml() {
		return html;
	}

	public void setHtml(boolean html) {
		this.html = html;
	}

    @ReportField(type = FieldType.Boolean)
	public boolean isTranslated() {
		return translated;
	}

	public void setTranslated(boolean translated) {
		this.translated = translated;
	}

	@Transient
	public static final Set<Integer> VALID_DEACTIVATED_EMAILS() {
		return PRIVATE_VALID_DEACTIVATED_EMAILS;
	}

	@Transient
	@Override
	public String getI18nKey() {
		return this.getClass().getSimpleName() + "." + id;
	}

	@Transient
	@Override
	public String getI18nKey(String property) {
		return getI18nKey() + "." + property;
	}

	public void cascadeRequiredLanguages(List<String> add, List<String> remove) {
		return;
	}

	public boolean hasMissingChildRequiredLanguages() {
		return getLanguages().isEmpty();
	}
}