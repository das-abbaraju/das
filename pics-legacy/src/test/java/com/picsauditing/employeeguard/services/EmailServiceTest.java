package com.picsauditing.employeeguard.services;

import com.picsauditing.dao.EmailTemplateDAO;
import com.picsauditing.dao.TokenDAO;
import com.picsauditing.employeeguard.entities.builders.EmailHashBuilder;
import com.picsauditing.employeeguard.entities.builders.SoftDeletedEmployeeBuilder;
import com.picsauditing.i18n.service.TranslationService;
import com.picsauditing.jpa.entities.EmailQueue;
import com.picsauditing.jpa.entities.EmailTemplate;
import com.picsauditing.jpa.entities.ListType;
import com.picsauditing.mail.EmailSender;
import com.picsauditing.service.AppPropertyService;
import com.picsauditing.service.i18n.TranslationServiceFactory;
import com.picsauditing.util.SpringUtils;
import org.apache.commons.io.FileUtils;
import org.approvaltests.Approvals;
import org.approvaltests.reporters.DiffReporter;
import org.approvaltests.reporters.UseReporter;
import org.junit.Before;
import org.junit.Test;
import org.mockito.ArgumentCaptor;
import org.mockito.Mock;
import org.mockito.MockitoAnnotations;
import org.powermock.reflect.Whitebox;
import org.springframework.context.ApplicationContext;

import java.io.File;
import java.io.IOException;

import static org.junit.Assert.*;
import static org.mockito.Matchers.anyInt;
import static org.mockito.Matchers.anyString;
import static org.mockito.Mockito.*;

@UseReporter(DiffReporter.class)
public class EmailServiceTest {

	public static final int APP_USER_ID = 231;

	// Class under test
	private EmailService emailService;

	@Mock
	private ApplicationContext applicationContext;
	@Mock
	private AppPropertyService appPropertyService;
	@Mock
	private EmailSender emailSender;
	@Mock
	private EmailTemplateDAO emailTemplateDAO;
	@Mock
	private TokenDAO tokenDAO;
	@Mock
	private TranslationService translationService;

	@Before
	public void setUp() throws Exception {
		emailService = new EmailService();

		MockitoAnnotations.initMocks(this);

		Whitebox.setInternalState(emailService, "emailSender", emailSender);
		Whitebox.setInternalState(emailService, "appPropertyService", appPropertyService);
		Whitebox.setInternalState(SpringUtils.class, "applicationContext", applicationContext);
		Whitebox.setInternalState(TranslationServiceFactory.class, "translationService", translationService);

		when(applicationContext.getBean("EmailTemplateDAO")).thenReturn(emailTemplateDAO);
		when(applicationContext.getBean("TokenDAO")).thenReturn(tokenDAO);
	}

	@Test
	public void testSendEGWelcomeEmail() throws Exception {
		when(emailTemplateDAO.find(anyInt())).thenReturn(buildFakeEGWelcomeEmailTemplate());

		emailService.sendEGWelcomeEmail(new EmailHashBuilder()
				.hash("THIS_IS_THE_HASH_CODE")
				.employee(new SoftDeletedEmployeeBuilder().firstName("Bob The Employee").build())
				.build(),
				"Contractor Account");

		verifyTestSendEGWelcomeEmail();
	}

	private EmailTemplate buildFakeEGWelcomeEmailTemplate() throws IOException {
		EmailTemplate fakeEmailTemplate = new EmailTemplate();

		fakeEmailTemplate.setAllowsVelocity(true);
		fakeEmailTemplate.setHtml(true);
		fakeEmailTemplate.setListType(ListType.Audit);
		fakeEmailTemplate.setSubject("Welcome to PICS EmployeeGUARD");
		fakeEmailTemplate.setBody(FileUtils.readFileToString(new File("src/test/java/com/picsauditing" +
				"/employeeguard/services/EmployeeGUARDWelcomEmailTemplate.html")));

		return fakeEmailTemplate;
	}

	private void verifyTestSendEGWelcomeEmail() throws Exception {
		ArgumentCaptor<EmailQueue> emailQueue = ArgumentCaptor.forClass(EmailQueue.class);
		verify(emailSender).sendNow(emailQueue.capture());

		assertEquals("Welcome to PICS EmployeeGUARD", emailQueue.getValue().getSubject());
		Approvals.verify(emailQueue.getValue().getBody());
	}

	@Test
	public void testSendEGFeedBackEmail() throws Exception {
		when(emailTemplateDAO.find(anyInt())).thenReturn(buildFakeEGFeedBackEmailTemplate());

		boolean result = emailService.sendEGFeedBackEmail("Some feedback", "Contractor Account", APP_USER_ID, "user@email.com");

		verifyTestSendEGFeedBackEmail(result);
	}

	private void verifyTestSendEGFeedBackEmail(boolean result) throws Exception {
		assertTrue(result);

		ArgumentCaptor<EmailQueue> emailQueue = ArgumentCaptor.forClass(EmailQueue.class);
		verify(emailSender).sendNow(emailQueue.capture());

		assertEquals("EmployeeGUARD Feedback", emailQueue.getValue().getSubject());
		Approvals.verify(emailQueue.getValue().getBody());
	}

	@Test
	public void testSendEGFeedBackEmail_Exception() throws Exception {
		when(appPropertyService.getPropertyInt(anyString(), anyInt())).thenThrow(new RuntimeException());

		boolean result = emailService.sendEGFeedBackEmail("Some feedback", "Contractor Account", 231, "user@email.com");

		assertFalse(result);
		verifyZeroInteractions(emailSender);
	}

	private EmailTemplate buildFakeEGFeedBackEmailTemplate() {
		EmailTemplate fakeEmailTemplate = new EmailTemplate();

		fakeEmailTemplate.setAllowsVelocity(true);
		fakeEmailTemplate.setSubject("EmployeeGUARD Feedback");
		fakeEmailTemplate.setBody(" Account Name: ${accountName}\r\n appUserId: ${appUserId}\r\n " +
				"User Email Address: ${userEmailAddress}\r\n Feedback: ${feedback}\r\n\r\n");

		return fakeEmailTemplate;
	}
}
