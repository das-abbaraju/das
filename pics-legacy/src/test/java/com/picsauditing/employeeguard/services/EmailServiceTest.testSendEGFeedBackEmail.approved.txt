 Account Name: Contractor Account
 appUserId: 231
 User Email Address: user@email.com
 Feedback: Some feedback

